standalone/include/psen_scan_v2_standalone/protocol_layer/scanner_state_machine_def.h

```diff
@@ -167,13 +167,37 @@ inline void ScannerProtocolDef::handleMonitoringFrame(const scanner_events::RawM
                              util::formatRange(frame.diagnosticMessages()));
     }
 
-    printUserMsgFor(complete_scan_validator_.validate(frame, DEFAULT_NUM_MSG_PER_ROUND));
-    if (frame.measurements().empty())
+    if (args_->config_.fragmentedScansEnabled())
     {
-      PSENSCAN_DEBUG("StateMachine", "No measurement data in this message, skipping laser scan callback.");
-      return;
+      printUserMsgFor(complete_scan_validator_.validate(frame, DEFAULT_NUM_MSG_PER_ROUND));
+      if (frame.measurements().empty())
+      {
+        PSENSCAN_DEBUG("StateMachine", "No measurement data in this message, skipping laser scan callback.");
+        return;
+      }
+      args_->inform_user_about_laser_scan_cb(data_conversion_layer::toLaserScan({ frame }));
+    }
+    else
+    {
+      if (!message_buffer_.empty() and message_buffer_[0].scanCounter() != frame.scanCounter())
+      {
+        PSENSCAN_WARN("StateMachine",
+                      "Detected dropped MonitoringFrame."
+                      " (Please check the ethernet connection or contact PILZ support if the error persists.)");
+        message_buffer_.clear();
+      }
+      message_buffer_.push_back(frame);
+      if (message_buffer_.size() == DEFAULT_NUM_MSG_PER_ROUND)
+      {
+        args_->inform_user_about_laser_scan_cb(data_conversion_layer::toLaserScan(message_buffer_));
+        message_buffer_.clear();
+      }
+      if (message_buffer_.size() > DEFAULT_NUM_MSG_PER_ROUND)
+      {
+        PSENSCAN_WARN("StateMachine", "Unexpected: Too many MonitoringFrames for one scan round received.");
+        message_buffer_.clear();
+      }
```

Comment:
This case can not happen if the message buffer gets cleared in 193.
And if it doesn't get cleared it would produce the warning in 182.

So maybe we should just omit this case. And add a check in 182 to ignore msgs with smaller scan counter to not ruin a new round if it already started.